README.md

```diff
@@ -34,46 +34,125 @@ For running some specific tests you can do this as following:
 Usage
 -----
 
-A simple workflow would be checking out a base repository,
-applying patches from Patchwork, and building the kernel.
+The `skt` tool implements several "commands", and each of those accepts its
+own command-line options and arguments. However there are also several
+"global" command-line options, shared by all the commands. To get a summary of
+the global options and available commands, run `skt.py --help`. To get a
+summary of particular command's options and arguments, run `skt.py <COMMAND>
+--help`, where `<COMMAND>` would be the command of interest.
+
+Most of command-line options can also be read by `skt` from its configuration
+file, which is `~/.sktrc` by default, but can also be specified using the
+global `--rc` command-line option. However, there are some command-line
+options which cannot be stored in the configuration file, and there are some
+options read from the configuration file by some `skt` commands, which cannot
+be passed via the command line. Some of the latter are required for operation.
+
+Most `skt` commands can write their state to the configuration file as they
+work, so that the other commands can take the workflow task over from them.
+Some commands can receive that state from the command line, via options, but
+some require some information stored in the configuration file. For this
+reason, to support a complete workflow, it is necessary to always make the
+commands transfer their state via the configuration file. That can be done by
+passing the global `--state` option with every command.
+
+To separate the actual configuration from the specific workflow's state, and
+to prevent separate tasks from interfering with each other, you can store your
+configuration in a separate (e.g. read-only) file, copy it to a new file each
+time you want to do something, then discard the file after the task is
+complete. Note that reusing a configuration file with state added can break
+some commands in unexpected ways. That includes repeating a previous command
+after the next command in the workflow has already ran.
```

Comment:
It is a state-config file like the light is a wave-particle :man_facepalming:. Sorry I couldn't avoid it

Reply:
Ha-ha, indeed :smile:. We have a FIXME somewhere, but now it's in #27 as well.